src/peer.rs

```diff
@@ -88,6 +90,86 @@ pub(crate) async fn server_peer_loop(peer_id: PeerId, broker_writer: Sender<Brok
             let password = login.get("password").ok_or("Password login param is missing")?.as_str();
             let login_type = login.get("type").map(|v| v.as_str()).unwrap_or("");
 
+            if login_type == "AZURE" {
+                let client = GraphClient::new(password);
+
+                let me_response = client
+                    .me()
+                    .get_user()
+                    .send()
+                    .await?;
+                let me_json: serde_json::Value = me_response.json().await?;
+                user = me_json
+                    .as_object()
+                    .and_then(|user_obj| user_obj.get("mail"))
+                    .and_then(|mail| mail.as_str())
+                    .map(|mail| "azure:".to_string() + mail)
+                    .ok_or("Azure: unexpected `me` response")?;
+
+                let groups_response = client
+                    .me()
+                    .transitive_member_of()
+                    .list_transitive_member_of()
+                    .send()
+                    .await?;
+
+                let groups_json: serde_json::Value = groups_response.json().await?;
+                let groups_from_azure = groups_json
+                    .as_object()
+                    .and_then(|obj| obj.get("value"))
+                    .and_then(|value_obj| value_obj.as_array())
+                    .map(|group_array| group_array
+                        .iter()
+                        .filter_map(|group_array_values| group_array_values.as_object())
+                        .filter(|value| value
+                            .get("@odata.type")
+                            .and_then(|x| x.as_str())
+                            .is_some_and(|data_type| data_type == "#microsoft.graph.group"))
+                        .filter_map(|group| group
+                            .get("id")
+                            .and_then(|id| id.as_str())
+                            ))
+                    .ok_or("Azure: unexpected error response")?;
+
+                let mut mapped_groups = groups_from_azure
+                    .map(|azure_group_uuid| async {
+                        match broker_writer.send(BrokerCommand::GetAzureGroupMapping { sender: peer_writer.clone(), group: azure_group_uuid.to_string() }).await {
```

Comment:
The `FuturesOrdered` could be avoided if `BrokerCommand::GetAzureGroupMapping` accepted a list of groups instead of a single group and `BrokerToPeerMessage::AzureMappingGroups` could then return mappings for each requested Azure group on that single command.